### PR TITLE
Add new subctl aws cloud commands

### PR DESCRIPTION
**Pull request by tpantelis**

### CODE REVIEW

**Review by skitt — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Review by Jaanki — CHANGES_REQUESTED** [review]
Also I am thinking if there should be separate packages for `cleanup` and `prepare` under `cmd/subctl/cloud` like how it is currently implemented.

File: cmd/subctl/aws.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		},
+	}
+
+	awsCleanupCmd = &cobra.Command{
+		Use:   "aws",
+		Short: "Clean up an AWS cloud",
+		Long: "This command cleans up an OpenShift installer-provisioned infrastructure (IPI) on AWS-based" +
+			" cloud after Submariner uninstallation.",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := cleanup.AWS(&restConfigProducer, cli.NewReporter())
```
Comment (Jaanki) [inline cmd/subctl/aws.go]: `cleanup` also needs `infraid` and `region` flags just like `prepare` and those value would need to be passed on.

**Review by Jaanki — CHANGES_REQUESTED** [review]

File: cmd/subctl/aws.go (hunk 1 above)
Comment (tpantelis) [reply cmd/subctl/aws.go]: Done

**Comment by tpantelis** [comment]
> Also I am thinking if there should be separate packages for `cleanup` and `prepare` under `cmd/subctl/cloud` like how it is currently implemented.

The bulk of the actual code is under `pkg/cloud` so `cmd/subctl/cloud` is basically just setting up cobra commands so I think separate packages would be overkill. Also it would require making some things public and explicitly referencing the packages in `main` so they're initialized which, again, I don't think is worth it.

File: pkg/subctl/cmd/cloud/cleanup/aws.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ func newAWSCleanupCommand(restConfigProducer restconfig.Producer) *cobra.Command
 		Run: func(cmd *cobra.Command, args []string) {
 			status := cli.NewReporter()
 
-			err := cleanup.AWS(&restConfigProducer, status)
+			err := cleanup.AWS(&restConfigProducer, &config, status)
```
Comment (Jaanki) [inline pkg/subctl/cmd/cloud/cleanup/aws.go:43]: The `config` struct fields `infraID, region, profile, credentialsFile, ocpMetadataFile` need to populated from the flag. And `config.GwInstanceType` to be passed as empty string.

**Review by Jaanki — CHANGES_REQUESTED** [review]

File: pkg/subctl/cmd/cloud/cleanup/aws.go (hunk 2 above)
Comment (tpantelis) [reply pkg/subctl/cmd/cloud/cleanup/aws.go:43]: That's the case from what I see. What do you see different?

Comment (Jaanki) [reply pkg/subctl/cmd/cloud/cleanup/aws.go:43]: I would have explicitly made `config.GwInstanceType = ""`

File: cmd/subctl/cloud_cleanup.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/*
```
Comment (Jaanki) [inline cmd/subctl/cloud_cleanup.go]: How about this is merged with the above file `cloud.go`

File: cmd/subctl/cloud_prepare.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/*
```
Comment (Jaanki) [inline cmd/subctl/cloud_prepare.go]: Ditto

**Review by Jaanki — COMMENTED** [review]
How about we have 1 file to add cloud, clean and prepare and then individual files for each platform
```
cmd/subctl
  |
  |- cloud.go -> add cloud, prepare and clean command
  |- aws.go
  |- rhos.go
```
and so on

File: pkg/subctl/cmd/cloud/cleanup/aws.go (hunk 2 above)
Comment (tpantelis) [reply pkg/subctl/cmd/cloud/cleanup/aws.go:43]: No need - empty string is the default.

**Review by Jaanki — APPROVED** [review]

**Pull request merged**